### PR TITLE
implemented assert.doesNotMatch

### PR DESCRIPTION
**Pull request by markusn**
### What does this PR do?

It implements assert.doesNotMatch making node:assert fully implemented.

_The documentation and TypeScript definitions already exists, this just provides the missing implementation._

- [ ] Documentation or TypeScript types (it's okay to leave the rest blank in this case)
- [x] Code changes

### How did you verify your code works?

I wrote automated tests.

If JavaScript/TypeScript modules or builtins changed:

- [x] I included a test for the new code, or existing tests cover it
- [x] I ran my tests locally and they pass (`bun-debug test test-file-name.test`)

If new methods, getters, or setters were added to a publicly exposed class:

_The documentation and TypeScript definitions already exists, this just provides the missing implementation._

- [ ] I added TypeScript types for the new methods, getters, or setters

### CODE REVIEW

File: test/js/node/assert/assert-doesNotMatch.test.cjs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+test("doesNotMatch does not throw when not matching", () => {
+  assert.doesNotMatch('I will pass', /different/);
+});
+
+test("doesNotMatch throws when argument is not string", () => {
+  try {
+    assert.doesNotMatch(123, /pass/);
+    expect.unreachable();
+  } catch (e) {}
```
Comment (paperclover) [inline test/js/node/assert/assert-doesNotMatch.test.cjs]: check that the type of error thrown here is not an expect unreachable error

File: test/js/node/assert/assert-doesNotMatch.test.cjs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+});
+
+test("doesNotMatch throws when matching", () => {
+  try {
+    assert.doesNotMatch('I will fail', /fail/);
+    expect.unreachable();
+  } catch (e) {}
```
Comment (paperclover) [inline test/js/node/assert/assert-doesNotMatch.test.cjs]: check error

**Comment by markusn** [comment]
@Jarred-Sumner I've looked at the failing tests and they seem unrelated to the changes in this PR.

**Review by paperclover — CHANGES_REQUESTED** [review]

**Comment by paperclover** [comment]
i apologize, these review comments were written days ago but i hadnt pressed submit

File: test/js/node/assert/assert-doesNotMatch.test.cjs (hunk 1 above)
Comment (markusn) [reply test/js/node/assert/assert-doesNotMatch.test.cjs]: thanks for the comment, this actually highlighted an actual problem with the code

File: test/js/node/assert/assert-doesNotMatch.test.cjs (hunk 2 above)
Comment (markusn) [reply test/js/node/assert/assert-doesNotMatch.test.cjs]: thanks for the comment, this actually highlighted an actual problem with the code

**Comment by markusn** [comment]
> i apologize, these review comments were written days ago but i hadnt pressed submit

No problem, I've changed the tests to expect the error instead.

**Review by paperclover — APPROVED** [review]

**Comment by paperclover** [comment]
thank you

**Pull request merged**